libraries/AP_NavEKF2/AP_NavEKF2_Control.cpp

```diff
@@ -105,6 +105,7 @@ void NavEKF2_core::setWindMagStateLearningMode()
     bool setMagInhibit = !magCalRequested || magCalDenied;
     if (!inhibitMagStates && setMagInhibit) {
         inhibitMagStates = true;
+        resetMagBodyVariances();
```

Comment:
We shouldn't have to do this here is we are doing a reset in the CovariancePrediction.